tests/repro_tests/data_association/test_point3dinitializer_reproducibility.py

```diff
@@ -0,0 +1,120 @@
+"""Reproducibility tests for Point3D initializer, in different configurations.
+
+Authors: Ayush Baid
+"""
+import unittest
+from typing import Dict, Optional, Tuple
+
+import gtsam
+import numpy as np
+from gtsam import PinholeCameraCal3Bundler, SfmTrack
+
+import gtsfm.utils.io as io_utils
+import gtsfm.utils.tracks as track_utils
+from gtsfm.common.gtsfm_data import GtsfmData
+from gtsfm.common.sfm_track import SfmMeasurement, SfmTrack2d
+from gtsfm.data_association.point3d_initializer import (
+    Point3dInitializer,
+    TriangulationOptions,
+    TriangulationSamplingMode,
+    TriangulationExitCode,
+)
+from tests.repro_tests.test_repro_base import ReproducibilityTestBase
+
+# Load ground truth data with 3 images and 7 tracks.
+GTSAM_EXAMPLE_FILE: str = "dubrovnik-3-7-pre"
+GROUND_TRUTH_DATA: GtsfmData = io_utils.read_bal(gtsam.findExampleDataFile(GTSAM_EXAMPLE_FILE))
+
+MEASUREMENTS_NOISE_STANDARD_DEVIATION = 1  # in pixels
+
+TRIANGULATION_REPROJ_ERROR_THRESHOLD = 10  # using a high value to account for random noise in input measurements
+
+TRIANGULATION_RESULT_TYPE = Tuple[Optional[SfmTrack], Optional[float], TriangulationExitCode]
+
+
+class TestPoint3dInitializerNoRansac(ReproducibilityTestBase, unittest.TestCase):
+    """Reproducibility tests for point3d initializer with no RANSAC."""
+
+    def setUp(
+        self,
+        triangulation_options: TriangulationOptions = TriangulationOptions(
+            reproj_error_threshold=TRIANGULATION_REPROJ_ERROR_THRESHOLD, mode=TriangulationSamplingMode.NO_RANSAC
+        ),
+    ) -> None:
+        camera_dict: Dict[int, PinholeCameraCal3Bundler] = {
+            i: GROUND_TRUTH_DATA.get_camera(i) for i in range(GROUND_TRUTH_DATA.number_images())
+        }
+        self._point3d_initializer: Point3dInitializer = Point3dInitializer(camera_dict, triangulation_options)
+        track_2d_noisefree: SfmTrack2d = track_utils.cast_3dtrack_to_2dtrack(GROUND_TRUTH_DATA.get_track(0))
+
+        # add noise to all the points
+        self._track_2d: SfmTrack2d = SfmTrack2d(
+            measurements=[
+                SfmMeasurement(
+                    measurement.i, measurement.uv + MEASUREMENTS_NOISE_STANDARD_DEVIATION * np.random.randn(2)
+                )
+                for measurement in track_2d_noisefree.measurements
+            ]
+        )
+
+    def run_once(self) -> TRIANGULATION_RESULT_TYPE:
+        return self._point3d_initializer.triangulate(self._track_2d)
+
+    def assert_results(self, results_a: TRIANGULATION_RESULT_TYPE, results_b: TRIANGULATION_RESULT_TYPE) -> None:
+        self.__assert_track3d(results_a[0], results_b[0])
+        self.assertAlmostEqual(results_a[1], results_b[1])
+        self.assertEqual(results_a[2], results_b[2])
+
+    def __assert_track3d(self, track3d_a: Optional[SfmTrack], track3d_b: Optional[SfmTrack]) -> None:
```

Comment:
the fn name sounds a bit like we are asserting this is a track. i think we also want to assert they're equal tracks, right?
would `__assert_track3d_equal()` make sense here?

Reply:
or should we add this on the GTSAM side as a .equals() method?